src/Microsoft.Net.Http.Headers/DateTimeFormatter.cs

```diff
@@ -0,0 +1,161 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using static System.Text.Encoding;
+
+namespace Microsoft.Net.Http.Headers
+{
+    internal static class DateTimeFormatter
```

Comment:
I'm unsure of the level of optimization that's really required for this code, but a few off the cuff thoughts:
1. My intuition says there's gotta be a better way of copying the bytes than a `foreach`
2. It might be beneficial to 'pack' all of these bytes into an array + offset for locality
3. We should look into the inlinability of all of this (most related to 1)